src/search/SearchMessagesScreen.js

```diff
@@ -130,6 +154,50 @@ class SearchMessagesScreenInner extends PureComponent<Props, State> {
     this.handleQuerySubmit(e);
   };
 
+  fetchOlder = async () => {
+    if (
+      this.props.loading
+      || this.state.foundOldest
+      || this.state.isFetching
+      || this.isFetchingOlder
+    ) {
+      return;
+    }
+
+    invariant(this.state.messages !== null, 'must have atleast 1 message already fetched.');
+    this.isFetchingOlder = true;
+    try {
+      const { query } = this.state;
+      const anchor = this.state.messages[0].id;
+      const { messages: fetchedMessages, foundOldest } = await await this.props.dispatch(
```

Comment:
This `await await` looks real weird. :-) You noticed that too when you first wrote this code, and I looked into it:
https://github.com/zulip/zulip-mobile/pull/5052#issuecomment-960296275
and it seems to be some harmless Flow bug.

I'd forgotten all about that, though, and it caught my attention again. Let's add a comment so the next reader knows what we know:
```suggestion
      // FlowIssue: Flow insists on this `await await`.  Should be equivalent to `await`, so harmless.
      //   See: https://github.com/zulip/zulip-mobile/pull/5052#issuecomment-960296275
      const { messages: fetchedMessages, foundOldest } = await await this.props.dispatch(
```